pkg/datapath/linux/ethtool/ethtool.go

```diff
@@ -0,0 +1,83 @@
+package ethtool
+
+import (
+	"bytes"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+)
+
+const (
+	SIOCETHTOOL      = 0x8946
+	ETHTOOL_GDRVINFO = 0x00000003 // Get driver info
+
+	IFNAMSIZ = 16
+)
+
+const (
+	veth = "veth"
+)
+
+type ifreq struct {
+	name [IFNAMSIZ]byte
+	data uintptr
+}
+
+type ethtoolDrvInfo struct {
+	cmd         uint32
+	driver      [32]byte
+	version     [32]byte
+	fwVersion   [32]byte
+	busInfo     [32]byte
+	eromVersion [32]byte
+	reserved2   [12]byte
+	nPrivFlags  uint32
+	nStats      uint32
+	testInfoLen uint32
+	eedumpLen   uint32
+	regdumpLen  uint32
+}
+
+func ethtoolIoctl(iface string, info *ethtoolDrvInfo) error {
+	var ifname [IFNAMSIZ]byte
+
+	fd, err := unix.Socket(unix.AF_INET, unix.SOCK_DGRAM, unix.IPPROTO_IP)
+	if err != nil {
+		return err
+	}
+	copy(ifname[:], iface)
+	req := ifreq{
+		name: ifname,
+		data: uintptr(unsafe.Pointer(info)),
+	}
+	_, _, ep := unix.Syscall(unix.SYS_IOCTL, uintptr(fd), SIOCETHTOOL, uintptr(unsafe.Pointer(&req)))
+	if ep != 0 {
+		return ep
+	}
+	return nil
+}
+
+func GetDeviceName(iface string) (string, error) {
```

Comment:
I don't think we are using safchain/ethtool anywhere in Cilium, at least not in ./pkg, and would prefer to avoid pulling in another dependency if possible. Seems best to leave it here IMO.

Reply:
I'm inclined to agree with John here. I don't know why it's in go.sum, we don't use it anywhere:

```
$ git grep safchain
go.sum:github.com/safchain/ethtool v0.0.0-20190326074333-42ed695e3de8/go.mod h1:Z0q5wiBQGYcxhMZ6gUqHn6pYNLypFAvaL3UvgZLR0U4=
```